spec/std/weak_ref_spec.cr

```diff
@@ -0,0 +1,70 @@
+require "spec"
+
+class State
```

Comment:
`State` and `Foo` might eventually conflict with other specs. Maybe use names like `WeakRefState` and `WeakRefFoo`?

This is one of the reasons I want to implement file-private types.